bee-ledger/src/workers/snapshot/download.rs

```diff
@@ -1,41 +1,200 @@
 // Copyright 2020-2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::workers::snapshot::error::Error;
+use crate::{
+    types::snapshot::SnapshotHeader,
+    workers::snapshot::{config::DownloadUrls, error::Error},
+};
 
+use bee_common::packable::Packable;
+
+use bytes::Buf;
+use futures::{future::join_all, StreamExt};
 use log::{info, warn};
+use reqwest::Response;
+
+use std::{io::Read, path::Path};
+
+async fn download_snapshot_header(download_url: &str) -> Result<SnapshotHeader, Error> {
+    info!("Downloading snapshot header {}...", download_url);
```

Comment:
I wouldn't put this log as INFO, this could spam when having many sources and it's not very interesting to the node owner, maybe put it as debug.